src/compiler/checker.ts

```diff
@@ -12192,15 +12192,20 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             type = tryGetTypeFromEffectiveTypeNode(declaration) || checkObjectLiteralMethod(declaration, CheckMode.Normal);
         }
         else if (
-            isParameter(declaration)
-            || isPropertyDeclaration(declaration)
+            isPropertyDeclaration(declaration)
             || isPropertySignature(declaration)
             || isVariableDeclaration(declaration)
             || isBindingElement(declaration)
             || isJSDocPropertyLikeTag(declaration)
         ) {
             type = getWidenedTypeForVariableLikeDeclaration(declaration, /*reportErrors*/ true);
         }
+        else if (isParameter(declaration)) {
+            type = getWidenedTypeForVariableLikeDeclaration(declaration, /*reportErrors*/ true);
+            if (declaration.dotDotDotToken) {
+                type = normalizeNoInferSpread(type);

```

Comment:
Maybe I'm thinking about this wrong, but shouldn't this actually be a part of `expandParameters` or something? IIRC that's where we normalize these out. Compared to sprinkling it around elsewhere.

Reply:
`getExpandedParameters` is only used by `signatureToSignatureDeclarationHelper` and node builder. This change has behavioral changes for inference etc so having this `getExpandedParameters` wouldn't work